portfolio/src/main/java/com/google/sps/servlets/FormServlet.java

```diff
@@ -50,20 +50,37 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve the maximum quantity of comments to display
+    String radioValueString = request.getParameter("value");
+    int max = Integer.parseInt(radioValueString);
```

Comment:
Nit

This param come from the wild, in a real web application, it's possible (even likely) that people with dubious motives will be sending all kinds of garbage in the parameters. It's important to make no assumptions about the quality of parameters. Assume that if you're expecting a number from a string, you will receive values such as

* -1
* 0
* 'pizza'

So your application should sanity check any received parameters and fail gracefully. In most cases this means catch any error thrown from parsing, and return a 400 bad request.

Reply:
This makes sense! I originally left out this check because I thought the radio button nature of the input would prevent this, but I now realize that there might be other ways of messing with the parameters. I added a check to ensure that max is an int with a value of 3, 5, or 10 and returned an I/O error if not.